mitodl/mitxonline
# Adds support for tying a discount to a specific product

#### Pre-Flight checklist


#### What are the relevant tickets?

#687 

#### What's this PR do?

Adds support for tying a discount to a product. If the discount is tied to a product, it will only be redeemable if the learner's basket contains a product that the discount is associated with. 

#### How should this be manually tested?
Automated tests should pass.

1. Select a discount code and product to test with.
2. In Django Admin, add a new Discount Product record for the discount and product. 
3. Attempt to add that product to the cart. 
4. Then, apply the discount code. It should apply successfully.
5. Clear your cart and add a different item to the cart.
6. Apply the discount code again. You should get a message saying the code is invalid. 

Similarly, you should be able to add another Discount Product for the selected discount and a different product and the discounts should apply (or not) in the same manner.

You should also be able to load the discount in the staff dashboard and see any products that are assigned to the discount. 

#### Other

You can't assign products to discounts in the staff dashboard at this point. There are other things you can't do (like assign users) and other things it needs so I'd rather wrap all that up in a separate issue/PR.
